plugins/modules/system/osx_defaults.py

```diff
@@ -229,8 +229,8 @@ def _convert_defaults_str_to_list(value):
         value.pop(0)
         value.pop(-1)
 
-        # Remove extra spaces and comma (,) at the end of values
-        value = [re.sub(',$', '', x.strip(' ')) for x in value]
+        # Remove spaces at beginning and comma (,) at the end, unquote and unescape double quotes
+        value = [re.sub('^ *"?|"?,?$', '', x.replace('\\"', '"')) for x in value]
```

Comment:
Can extra spaces at the end (after the comma) not appear? Otherwise,
```suggestion
        value = [re.sub('^ *"?|"?,? *$', '', x.replace('\\"', '"')) for x in value]
```
might be better. (The original code also handled spaces after the comma.)